src/const_eval.rs

```diff
@@ -0,0 +1,63 @@
+use rustc::hir::def_id::DefId;
+use rustc::traits::Reveal;
+use rustc::ty::subst::Substs;
+use rustc::ty::{self, TyCtxt};
+
+use error::{EvalError, EvalResult};
+use lvalue::{Global, GlobalId, Lvalue};
+use rustc_const_math::ConstInt;
+use eval_context::{EvalContext, StackPopCleanup};
+
+pub fn eval_body<'a, 'tcx>(
+    tcx: TyCtxt<'a, 'tcx, 'tcx>,
+    (def_id, substs): (DefId, &'tcx Substs<'tcx>),
+) -> EvalResult<'tcx, ConstInt> {
```

Comment:
This doesn't seem right, see https://github.com/rust-lang/rust/blob/d1f1f86d7f5b0ee363b3d247624c01db3b1ef55f/src/librustc/ty/maps.rs#L936-L939.
It's fine to have it *only* work for integers, but the return type isn't `ConstInt`.